modules/navier_stokes/src/fvbcs/INSFVTKEDWallFunctionBC.C

```diff
@@ -82,14 +84,16 @@ INSFVTKEDWallFunctionBC::boundaryValue(const FaceInfo & fi) const
   if (y_plus <= 5.0) // sub-laminar layer
   {
     const auto laminar_value = 2.0 * weight * TKE * mu / std::pow(dist, 2);
-    return laminar_value.value();
+    // Additional zero term to make sure new derivatives are not introduced as y_plus changes
+    return laminar_value + 0 * _mu_t(makeElemArg(&_current_elem), state);
   }
   else if (y_plus >= 30.0) // log-layer
   {
     const auto turbulent_value = weight * _C_mu(makeElemArg(&_current_elem), state) *
                                  std::pow(std::abs(TKE), 1.5) /
                                  (_mu_t(makeElemArg(&_current_elem), state) * dist);
-    return turbulent_value.value();
+    // Additional zero term to make sure new derivatives are not introduced as y_plus changes
+    return turbulent_value + 0 * mu;
```

Comment:
Same for this one. This is adding unnecessary computational burden to the linear solver. Please add a switch between linear and nonlinear if this is needed for the nonlinear solver